src/BTCPayServer.Lightning.CLightning/CLightningClient.cs

```diff
@@ -285,6 +285,25 @@ public async Task<LightningInvoice[]> ListInvoices(ListInvoicesParams request, C
             return invoices.Select(ToLightningInvoice).ToArray();
         }
 
+        async Task<LightningPayment[]> ILightningClient.ListPayments(CancellationToken cancellation)
+        {
+            return await ListPayments(null, cancellation);
+            }
+
+        public async Task<LightningPayment[]> ListPayments(ListPaymentsParams request, CancellationToken cancellation)
+        {
+            var payments = await SendCommandAsync<CLightningPayment[]>("listpays", null, false, true, cancellation);
+            if (request != null)
+            {
+                // we need to filter client-side, because the listpays command does not support these filters
+                payments = payments.Where(payment => 
+                    ((request.IncludePending.HasValue && request.IncludePending.Value) || ToPaymentStatus(payment.Status) != LightningPaymentStatus.Pending) &&
```

Comment:
`(request.IncludePending.HasValue && request.IncludePending.Value)` can be rewritten as `request.IncludePending is true`